retrofit-converters/jaxb/src/main/java/retrofit2/converter/jaxb/JaxbRequestConverter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2.converter.jaxb;
+
+import java.io.IOException;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import okio.BufferedSink;
+import retrofit2.Converter;
+
+import static retrofit2.converter.jaxb.JaxbConverterFactory.IGNORE_UNRECOGNIZED_FIELDS;
+
+final class JaxbRequestConverter<T> implements Converter<T, RequestBody> {
+  final XMLOutputFactory xmlOutputFactory = XMLOutputFactory.newInstance();
+  final JAXBContext context;
+  final Class<T> type;
+
+  JaxbRequestConverter(JAXBContext context, Class<T> type) {
+    this.context = context;
+    this.type = type;
+  }
+
+  @Override public RequestBody convert(final T value) throws IOException {
+    return new RequestBody() {
+      @Override public MediaType contentType() {
+        return JaxbConverterFactory.XML;
+      }
+
+      @Override public void writeTo(BufferedSink sink) throws IOException {
+        try {
+          Marshaller marshaller = context.createMarshaller();
+          marshaller.setEventHandler(IGNORE_UNRECOGNIZED_FIELDS);
+
+          XMLStreamWriter xmlWriter = xmlOutputFactory.createXMLStreamWriter(
+              sink.outputStream(), JaxbConverterFactory.XML.charset().name());
+          marshaller.marshal(value, xmlWriter);
```

Comment:
What happens if somebody mutates the value while it's being written out?

Reply:
https://github.com/square/retrofit/compare/jakew/jaxb/2018-01-10#diff-b645eed3acd195172d51a4e5b51a9d7fR49 seems consistent with the other request body converters.

Reply:
That's already a problem between `enqueue` and when the converter is called.

That being said, I don't think any other converter streams serialization currently. Is there a reason to prefer this over writing to a `Buffer` and capturing a `ByteString`? It seems like we should be consistent.

Reply:
Consistent works for me. Lemme fix.

Reply:
>That's already a problem between enqueue and when the converter is called.

Good point! Is there any obstacle to making all of the converters encode directly to the stream, then?

Reply:
Fixed.